PurchasesTests/Support/BeginRefundRequestHelperTests.swift

```diff
@@ -59,100 +148,173 @@ class BeginRefundRequestHelperTests: XCTestCase {
-            callbackCalled = true
-            receivedResult = result
-        }
-
-        expect(callbackCalled).toEventually(beTrue())
-        let nonNilReceivedResult: Result<RefundRequestStatus, Error> = try XCTUnwrap(receivedResult)
+        let receivedStatus = try await helper.beginRefundRequest(forProduct: mockProductID)
         let expectedStatus = RefundRequestStatus.success
-        expect(nonNilReceivedResult).to(beSuccess { status in
-            expect(status) == expectedStatus
-        })
+
+        expect(receivedStatus) == expectedStatus
         expect(self.sk2Helper.verifyTransactionCalled).to(beTrue())
         expect(self.sk2Helper.refundRequestCalled).to(beTrue())
     }
 
-    func testBeginRefundReturnsSuccessOnStoreKitSuccess() throws {
+    func testBeginRefundReturnsSuccessOnStoreKitSuccess() async throws {
         guard #available(iOS 15.0, macCatalyst 15.0, *) else {
             throw XCTSkip("Required API is not available for this test.")
         }
 
         sk2Helper.maybeMockSK2Status = StoreKit.Transaction.RefundRequestStatus.success
 
-        var callbackCalled = false
-        var receivedResult: Result<RefundRequestStatus, Error>?
 
-        helper.beginRefundRequest(productID: mockProductID) { result in
-            callbackCalled = true
-            receivedResult = result
-        }
+        let receivedStatus = try await helper.beginRefundRequest(forProduct: mockProductID)
 
-        expect(callbackCalled).toEventually(beTrue())
-        let nonNilReceivedResult: Result<RefundRequestStatus, Error> = try XCTUnwrap(receivedResult)
         let expectedStatus = RefundRequestStatus.success
-        expect(nonNilReceivedResult).to(beSuccess { status in
-            expect(status) == expectedStatus
-        })
+        expect(receivedStatus) == expectedStatus
     }
 
-    func testBeginRefundReturnsFailureOnStoreKitRefundRequestFailure() throws {
+    func testBeginRefundReturnsFailureOnStoreKitRefundRequestFailure() async throws {
         guard #available(iOS 15.0, macCatalyst 15.0, *) else {
             throw XCTSkip("Required API is not available for this test.")
         }
 
         let expectedError = ErrorUtils.beginRefundRequestError(withMessage: "test")
         sk2Helper.maybeMockSK2Error = expectedError
 
-        var callbackCalled = false
-        var receivedResult: Result<RefundRequestStatus, Error>?
 
-        helper.beginRefundRequest(productID: mockProductID) { result in
-            callbackCalled = true
-            receivedResult = result
+        do {
+            _ = try await helper.beginRefundRequest(forProduct: mockProductID)
+            XCTFail("beginRefundRequestForProduct should have thrown error")
+        } catch {
+            expect(error).to(matchError(expectedError))
+        }
+    }
+
+    func testBeginRefundForEntitlementFailsOnCustomerInfoFetchFail() async throws {
+        guard #available(iOS 15.0, macCatalyst 15.0, *) else {
```

Comment:
This can use `AvailabilityChecks.iOS15APIAvailableOrSkipTest()` I think?

Reply:
that one continues for mac, tv and watch, and this method is only available for ios

Reply:
are you sure that this evaluates to false for `watchOS / tvOS / macOS`? These checks are kinda dumb, I think they'll evaluate to `true` because they're only checking API level but not OS compatibility, which is why we check both OS and API in relevant sections by doing `#if os(iOS)` + `if #available`. 

I might be wrong though. 

In this case it's irrelevant, though, since these tests are in the `StoreKitUnitTest` target, which compiles for iOS-only. Note that it's iOS-only because it needs a host app and I didn't figure out how to set a different host app per platform. 

On a different note, I think we should move this file over to the `StoreKitUnitTest` target to make it clearer that they'll only run in that test plan / target.